Tests/GRPCTests/VsockSocketTests.swift

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import EchoImplementation
+import EchoModel
+@testable import GRPC
+import NIOPosix
+import XCTest
+
+class VsockSocketTests: GRPCTestCase {
+  func testVsockSocket() throws {
+    try XCTSkipUnless(self.vsockAvailable(), "Vsock unavailable")
+    let group = NIOPosix.MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    defer {
+      XCTAssertNoThrow(try group.syncShutdownGracefully())
+    }
+
+    // Setup a server.
+    let server = try Server.insecure(group: group)
+      .withServiceProviders([EchoProvider()])
+      .withLogger(self.serverLogger)
+      .bind(vsockAddress: .init(cid: .any, port: 31234))
+      .wait()
+    defer {
+      XCTAssertNoThrow(try server.close().wait())
+    }
+
+    let channel = try GRPCChannelPool.with(
+      target: .vsockAddress(.init(cid: .local, port: 31234)),
+      transportSecurity: .plaintext,
+      eventLoopGroup: group
+    )
+    defer {
+      XCTAssertNoThrow(try channel.close().wait())
+    }
+
+    let client = Echo_EchoNIOClient(channel: channel)
+    let resp = try client.get(Echo_EchoRequest(text: "Hello")).response.wait()
+    XCTAssertEqual(resp.text, "Swift echo get: Hello")
+  }
+
+  private func vsockAvailable() -> Bool {
+    let fd: CInt
+    #if os(Linux)
+    fd = socket(AF_VSOCK, CInt(SOCK_STREAM.rawValue), 0)
+    #elseif canImport(Darwin)
+    fd = socket(AF_VSOCK, SOCK_STREAM, 0)
+    #else
+    fd = -1
+    #endif
+    guard fd != -1 else { return false }
```

Comment:
This is a pet-peeve of mine: I think `guard !x else { ... }` is much less clear than `if x { ... }` (I appreciate they aren't exactly the same as `guard` forces you to return/throw)

```suggestion
    if fd == -1 { return false }
```

Reply:
Happy to accommodate the peev and have updated to use `if`.

OOI, don't you ever like to use the `guard` and it's force-you-to-return/throw semantics? I like the compiler to help me where possible. Was it the double-negative that triggered you? Would you have been happier with `guard fd >= 0 else { ... }` or always prefer `if <exit condition> { ... }`?